### PR TITLE
Fixed typo and version mismatch in upgrade guide Volto 17

### LINKED ISSUE #5702: Typo and version mismatch in upgrade guide Volto 17
```
"devDependencies": {
+     "@plone/scripts": ^3.0.0,
+     "@typescript-eslint/eslint-plugin": "6.7.0",
+     "@typescript-eslint/parser": "6.7.0",
+     "stylelint-prettier": "1.1.2",
+     "ts-jest": "^26.4.2",
+     "ts-loader": "9.4.4",
+     "typescript": "5.2.2"
}
```

### PR DESCRIPTION
**Pull request by ksuess**
Fix #5702

### CODE REVIEW

**Review by ichim-david — APPROVED** [review]

**Comment by stevepiercy** [comment]
First @ksuess is correct to update this discrepancy. But I have to ask, is there a better way that reduces maintenance of the docs and gives developers an easier way to update their code? Why, yes, there is.

I chatted with @ichim-david in Discord. I think it's too much work to maintain dependencies as diffs in documentation every time we update the deps in Volto. Additionally copy-pasting a diff into one's `package.json` is not efficient. And the diffs in the docs conflict with one another, as @ksuess shows in this PR.

It is easier to go to the source file in the repo for that branch, copy-paste that into your `package.json`, then run a diff to see what has changed. When I want to update my dev deps for a 17.x.x site, I go here:

https://github.com/plone/volto/blob/17.x.x/packages/generator-volto/generators/app/templates/package.json.tpl#L139-L161

Select "Copy lines" from the stoplight menu, and paste into my `package.json`. Diff, tweak, and done.

I'd suggest we remove the diffs, and instead point to the source code in the repo and add text, such as the following.

---

Dependencies in Volto 17.x.x will continue to be updated after this change note is published.
To update your project against the current release of Volto 17, you should compare the development dependencies in your `package.json` against those in [Volto's `17.x.x` branch's `package.json`](https://github.com/plone/volto/blob/17.x.x/packages/generator-volto/generators/app/templates/package.json.tpl#L139-L161).

**Comment by sneridagh** [comment]
@stevepiercy the upgrade guide ones are correct, and since the docs are "frozen" in time, they are correct. They are not going to change overtime without another breaking.

The others (the ones in the typescript.md) it's arguable they should point to the current latest stable generator, which is tested per version right now, so I guess it's ok.

**Comment by stevepiercy** [comment]
> @stevepiercy the upgrade guide ones are correct, and since the docs are "frozen" in time, they are correct. They are not going to change overtime without another breaking.

@sneridagh do you mean that @ksuess should revert the changes she made to `docs/source/upgrade-guide/index.md`? A code review would make it clear.

> The others (the ones in the typescript.md) it's arguable they should point to the current latest stable generator, which is tested per version right now, so I guess it's ok.

OK, so @ksuess should replace the diff in `docs/source/contributing/typescript.md` with the [text in my comment below the horizontal rule](https://github.com/plone/volto/pull/5703#issuecomment-1913102120), correct?

**Comment by sneridagh** [comment]
@stevepiercy yes, if possible.

File: docs/source/upgrade-guide/index.md [hunk 1]
```diff
@@ -148,10 +148,10 @@ Edit your {file}`package.json`:
 
 ```diff
 "devDependencies": {
-+     "@plone/scripts": ^3.0.0,
++     "@plone/scripts": "^3.0.0",
```
Comment (ksuess) [inline docs/source/upgrade-guide/index.md:151]: Quotation marks are missing.

**Review by ksuess — COMMENTED** [review]
You want me to revert these two changes in upgrade guide?
https://6.docs.plone.org/volto/upgrade-guide/index.html#upgraded-linters-eslint-prettier-and-stylelint

grafik

**Comment by sneridagh** [comment]
@stevepiercy I'm +1 to proceed from now on that way, pointing people to code instead of diffing in docs, let's do it for next contributions, ok? @ksuess PR is fine, as long as I can see and I'd merge it as it is.

**Pull request closed without merging**